framework/include/systems/SystemBase.h

```diff
@@ -232,12 +237,19 @@ class SystemBase : public libMesh::ParallelObject, public ConsoleStreamInterface
 
   /**
    * Get a state of the solution (0 = current, 1 = old, 2 = older, etc).
-   *
-   * By default, up to state _default_solution_states is added. Any older states must be
-   * added using the non-const solutionState().
    */
   const NumericVector<Number> & solutionState(const unsigned int state) const;
 
+  /**
+   * Registers that the solution state \p state is needed.
+   */
+  void needSolutionState(const unsigned int state);
```

Comment:
Can we use enum for the argument?

Reply:
Take a look here: https://github.com/idaholab/moose/pull/16363#discussion_r551435071

Reply:
duplicated question. I am ok to use unsigned int, just add more comments here like
```
0 - only current solution is needed
1- both current and old solutions are needed
2 - current, old and older solutions are needed
```

Reply:
Yeah, I definitely agree. I'll make it clearer. thanks!